File: core/src/nl/t64/game/rpg/screens/world/mapobjects/GameMapQuestTexture.java
package nl.t64.game.rpg.screens.world.mapobjects;

import com.badlogic.gdx.maps.objects.TextureMapObject;
import lombok.Getter;
import nl.t64.game.rpg.Utils;


public class GameMapQuestTexture {

    @Getter
    private final TextureMapObject texture;
    private final String questId;
    private final boolean isVisibleIfComplete;
    private final String taskId;
    @Getter
    private boolean isVisible;

    public GameMapQuestTexture(TextureMapObject textObject) {
        this.texture = textObject;
        this.questId = textObject.getName();
        this.isVisibleIfComplete = textObject.getProperties().get("visibleIfComplete", Boolean.class);
        this.taskId = textObject.getProperties().get("task", String.class);
        this.isVisible = textObject.getProperties().get("isVisible", Boolean.class);
    }

    public void update() {
        boolean isFinished = Utils.getGameData().getQuests().getQuestById(questId).isFinished();
        boolean isComplete = Utils.getGameData().getQuests().getQuestById(questId).isTaskComplete(taskId);
        isVisible = (isFinished || isComplete) == isVisibleIfComplete;
    }

}

File: core/src/nl/t64/game/rpg/screens/world/mapobjects/GameMapQuestTexture.kt
package nl.t64.game.rpg.screens.world.mapobjects

import com.badlogic.gdx.maps.objects.TextureMapObject
import ktx.tiled.property
import nl.t64.game.rpg.Utils.gameData
import nl.t64.game.rpg.components.quest.QuestState


class GameMapQuestTexture(val texture: TextureMapObject) {

    private val questId: String = texture.name
    private val isVisibleIfComplete: Boolean = texture.property("visibleIfComplete")
    private val taskId: String = texture.property("task")
    var isVisible: Boolean = texture.property("isVisible")

    fun update() {
        val isFinished = gameData.quests.isCurrentStateEqualOrHigherThan(questId, QuestState.FINISHED)
        val isComplete = gameData.quests.isTaskComplete(questId, taskId)
        isVisible = (isFinished || isComplete) == isVisibleIfComplete
    }

}
